src/core/client/framework/lib/bootstrap/TalkContext.tsx

```diff
@@ -62,6 +63,9 @@ export interface TalkContext {
 
   /** Clear session data. */
   clearSession: () => Promise<void>;
+
+  /** Controls router transitions (for tests) */
+  transitionControl?: TransitionControlData;
```

Comment:
Is it possible to protect access against this during production with polymorphism?

```ts
type A<T extends boolean> = { a: string } & (T extends true
  ? { b: string }
  : {});

// no error
const a: A<false> = { a: "123" };

// error
const b: A<true> = { a: "123" };
```

Reply:
Possibly, but not straight forward. Couldn't find a straight forward way to pull this off on first glance. I would also say, as it is marked as optional, you can't accidentally access it without checking for its existing or forcing it with `!`, which probably would have made you think twice about what this is for and when it is available.